marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.hub/lib/job-lib.xqy

```diff
@@ -0,0 +1,71 @@
+xquery version "1.0-ml";
+
+module namespace job = "http://marklogic.com/data-hub/job-lib";
+
+import module namespace config = "http://marklogic.com/data-hub/config"
+  at "/com.marklogic.hub/lib/config.xqy";
+
+declare option xdmp:mapping "false";
+
+declare variable $SUCCESS-KEY := "success";
+declare variable $FAILED-KEY  := "failed";
+
+(:
+ : Given a list of job-ids, delete those jobs and any traces associated with them.
+ :)
+declare function job:delete-jobs-and-traces($job-ids as xs:string*)
+{
+  let $job-results := job:delete-jobs($job-ids)
+  let $deleted-traces :=
+    xdmp:invoke-function(
+      function() {
+        job:delete-traces($job-ids)
+      },
+      <options xmlns="xdmp:eval">
+        <database>{xdmp:database($config:TRACE-DATABASE)}</database>
+      </options>
+    )
+  return
+    document {
+      object-node {
+        "totalCount": fn:count(map:get($job-results, $SUCCESS-KEY)),
+        "errorCount": fn:count(map:get($job-results, $FAILED-KEY)),
+        "deletedJobs": json:to-array(map:get($job-results, $SUCCESS-KEY)),
+        "deletedTraces": json:to-array($deleted-traces),
+        "failedJobs": json:to-array(map:get($job-results, $FAILED-KEY)),
+        "failedTraces": json:array(),
+        "errors": json:array()
+      }
+    }
+};
+
+declare private function job:delete-jobs($job-ids as xs:string*) as map:map
+{
+  let $results := map:map()
+  let $_ :=
+    for $id in $job-ids
+    let $uri := cts:uris((), (), cts:json-property-value-query("jobId", $id))
```

Comment:
can this be done like so?

```xquery
for $uri in cts:uris((), (), cts:json-property-value-query("jobId", $job-ids))
```

Reply:
I'm also using the $id in reporting back in the $results map, so I do need $id

Reply:
ok. cool.